src/coprocessor/dag/mod.rs

```diff
@@ -17,26 +17,20 @@ use crate::storage::{Statistics, Store};
 pub fn build_handler<S: Store + 'static>(
     req: DagRequest,
     ranges: Vec<KeyRange>,
-    start_ts: u64,
     store: S,
     data_version: Option<u64>,
     deadline: Deadline,
     batch_row_limit: usize,
     is_streaming: bool,
     enable_batch_if_possible: bool,
 ) -> Result<Box<dyn RequestHandler>> {
-    let mut is_batch = false;
-    if enable_batch_if_possible && !is_streaming {
-        let is_supported =
-            tidb_query::batch::runner::BatchExecutorsRunner::check_supported(req.get_executors());
-        if let Err(e) = is_supported {
-            // Not supported, will fallback to normal executor.
-            // To avoid user worries, let's output success message.
-            debug!("Successfully use normal Coprocessor query engine"; "start_ts" => start_ts, "reason" => %e);
-        } else {
-            is_batch = true;
-        }
-    }
+    // TODO: support batch executor while handling server-side streaming requests
+    let is_batch = if enable_batch_if_possible && !is_streaming {
+        tidb_query::batch::runner::BatchExecutorsRunner::check_supported(req.get_executors())?;
+        true
```

Comment:
This is too ugly 🤣You are writing: `let x = if a { true } else { false }`. Why not `let x = a`.